src/providers/cortex/trulens/providers/cortex/provider.py

```diff
@@ -32,15 +34,47 @@ def __init__(
             self.DEFAULT_MODEL_ENGINE if model_engine is None else model_engine
         )
         self_kwargs["endpoint"] = CortexEndpoint(*args, **kwargs)
+        
+        connection_parameters = {
+            "account": os.environ["SNOWFLAKE_ACCOUNT"],
+            "user": os.environ["SNOWFLAKE_USER"],
+            "role": os.environ["SNOWFLAKE_ROLE"],
+            "database": os.environ["SNOWFLAKE_DATABASE"],
+            "schema": os.environ["SNOWFLAKE_SCHEMA"],
+            "warehouse": os.environ["SNOWFLAKE_WAREHOUSE"],
+        }
+
+        if "SNOWFLAKE_USER_PASSWORD" in os.environ:
+            connection_parameters["password"] = os.environ[
+                "SNOWFLAKE_USER_PASSWORD"
+            ]
+
+        if "SNOWFLAKE_PRIVATE_KEY" in os.environ:
```

Comment:
`elif`? Also should this order be reversed? First give priority to key based auth?